runtime/lua/vim/lsp.lua

```diff
@@ -260,10 +262,19 @@ end
 --- `ftplugin/<filetype_name>.lua` (See |ftplugin-name|)
 ---
 --- @param config vim.lsp.ClientConfig Configuration for the server.
---- @param opts vim.lsp.start.Opts? Optional keyword arguments
+--- @param opts vim.lsp.start.Opts? Optional keyword arguments.
 --- @return integer? client_id
 function lsp.start(config, opts)
   opts = opts or {}
+
+  if opts.reuse_client == false then
+    local client_id, err = lsp._create_and_initialize_client(config)
+    if err and not opts.silent then
+      vim.notify(err, vim.log.levels.WARN)
+    end
+    return client_id
+  end
+
```

Comment:
If I read `lsp.start(config, { reuse_client = false })` I'd interpret that to mean "never re-use a client" and not make the connection that it influences the buffer attachment.

I think this should be something like `lsp.start(config, { attach = false })` instead. With the default reuse_client that would either be a no-op or start the client and server.

Reply:
Hmm good point.